packages/react-utils/src/compose/resolveShorthandProps.tsx

```diff
@@ -6,14 +6,17 @@ import * as React from 'react';
  * @param props - The incoming props
  * @param shorthandPropNames - An array of prop names to apply simplification to
  */
-export const resolveShorthandProps = <TProps,>(props: TProps, shorthandPropNames: (keyof TProps)[]) => {
+export const resolveShorthandProps = <TProps,>(props: TProps, shorthandPropNames: string[]) => {
   let newProps = props;
 
   if (shorthandPropNames && shorthandPropNames.length) {
     newProps = {
       ...props,
     };
     for (const propName of shorthandPropNames) {
+      // TODO find clean way of guaranteeing only shorthand props are typechecked
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
**suggestion:** this can be mitigated by setting proper constraints on the function API

Following will remove need of exposing `any` in our codebase 

```ts
const resolveShorthandProps = <TProps extends Record<string, unknown>>(
  props: TProps,
  shorthandPropNames: string[],
): TProps => { ... }
```